commit 4ceecbeaebc6b2e7b3449022f65b07f14580ccec (refs/changes/23/2923/1)
Author: Julien Gilli <julien.gilli@joyent.com>
Date:   2017-11-07T13:44:35-08:00 (1 year, 11 months ago)
    
    TOOLS-1923 VOLAPI node-sdc-clients tests should skip instead of exiting when VOLAPI not present

diff --git a/test/runtests b/test/runtests
index 01453f5..9c43144 100755
--- a/test/runtests
+++ b/test/runtests
@@ -116,6 +116,7 @@ if [[ -n "$opt_headnode" ]]; then
     export AMON_IP=$(ssh $opt_headnode 'bash /lib/sdc/config.sh -json | json amon_admin_ips | cut -d, -f1')
     export CNAPI_IP=$(ssh $opt_headnode 'bash /lib/sdc/config.sh -json | json cnapi_admin_ips | cut -d, -f1')
     export CA_IP=$(ssh $opt_headnode 'bash /lib/sdc/config.sh -json | json ca_admin_ips | cut -d, -f1')
+    export IMGAPI_IP=$(ssh $opt_headnode 'bash /lib/sdc/config.sh -json | json imgapi_admin_ips | cut -d, -f1')
     export NAPI_IP=$(ssh $opt_headnode 'bash /lib/sdc/config.sh -json | json napi_admin_ips | cut -d, -f1')
     export VMAPI_IP=$(ssh $opt_headnode 'bash /lib/sdc/config.sh -json | json vmapi_admin_ips | cut -d, -f1')
     export UFDS_ADMIN_UUID=$(ssh $opt_headnode 'bash /lib/sdc/config.sh -json | json ufds_admin_uuid')
diff --git a/test/volapi.test.js b/test/volapi.test.js
index 61a6e13..40cdbf1 100644
--- a/test/volapi.test.js
+++ b/test/volapi.test.js
@@ -12,14 +12,21 @@ var assert = require('assert-plus');
 var bunyan = require('bunyan');
 var libuuid = require('libuuid');
 var restify = require('restify-clients');
-var test = require('tape');
+var tapeTest = require('tape');
 
 assert.ok(typeof (process.env.NAPI_IP) === 'string' &&
     process.env.NAPI_IP !== '', 'NAPI_IP env var must be a non-empty string');
 
-if (!process.env.VOLAPI_IP) {
-    console.log('Could not find VOLAPI core service IP address, skipping.');
-    process.exit(0);
+function testIfVolapi(testName, testFunc) {
+    assert.string(testName, 'testName');
+    assert.func(testFunc, 'testFunc');
+
+    if (!process.env.VOLAPI_IP) {
+        console.log('# Could not find VOLAPI core service IP address, ' +
+            'skipping.');
+    } else {
+        tapeTest(testName, testFunc);
+    }
 }
 
 var NAPI = require('../lib/index').NAPI;
@@ -38,7 +45,7 @@ var VOLAPI_URL = 'http://' + VOLAPI_IP;
 
 var volApiClient;
 
-test('volapi', function (tt) {
+testIfVolapi('volapi', function (tt) {
     var volumeResOwnerUuid;
     var volumeResUuid;
 
